backend/de.metas.business/src/main/java/de/metas/shipping/PurchaseOrderToShipperTransportationService.java

```diff
@@ -185,4 +195,36 @@ else if (existingPackagesCount < qtyLUs)
 		}
 	}
 
+	@Nullable
+	public ReportResultData printSSCC18_Labels(
+			@NonNull final Properties ctx,
+			@NonNull final OrderId orderId)
+	{
+		final ImmutableList<PackageId> packageIDs = repo.getPackageIDsByOrderId(orderId);
+
+		Check.assumeNotEmpty(packageIDs, "packageIDs not empty");
+
+		//
+		// Create the process info based on AD_Process and AD_PInstance
+		final ProcessExecutionResult result = ProcessInfo.builder()
+				.setCtx(ctx)
+				.setAD_ProcessByValue(AD_PROCESS_VALUE_C_Order_SSCC_Print)
+				//
+				// Parameter: REPORT_SQL_QUERY: provide a different report query which will select from our datasource instead of using the standard query (which is M_HU_ID based).
+				.addParameter(ReportConstants.REPORT_PARAM_SQL_QUERY, "select * from report.fresh_M_Package_SSCC_Label_Report"
```

Comment:
if we have to change the report query, what's the benefit of reusing C_Order_SSCC_Print_Jasper instead of having a new process?

Reply:
C_Order_SSCC_Print_Jasper is a new process introduced by me on the model of EDI_DesadvLine_SSCC_Print which is also printing labels in the absence of HUs.

Reply:
The original query is the one written in the jasper: https://github.com/metasfresh/metasfresh/blob/new_dawn_uat/backend/de.metas.fresh/de.metas.fresh.base/src/main/jasperreports/de/metas/docs/label/sscc/label.jrxml#L30

```sql
 "select * from report.fresh_HU_SSCC_Label_Report where M_HU_ID="+$P{M_HU_ID}
 ```



In my process, I am filtering by C_Order_ID : https://github.com/metasfresh/metasfresh/blob/0e2c9db32abd90f39c2432b51a0c00078d5a53ae/backend/de.metas.business/src/main/java/de/metas/shipping/PurchaseOrderToShipperTransportationService.java#L211